codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/EventSourcedEntityTestKitGenerator.scala

```diff
@@ -74,13 +77,21 @@ object EventSourcedEntityTestKitGenerator {
           |
           |$imports
           |
-          |public class ${testkitClassName} {
+          |public final class ${testkitClassName} {
           |
           |  private $stateClassName state;
           |  private $entityClassName entity;
           |  private List<Object> events = new ArrayList<Object>();
           |  private AkkaServerlessTestKitHelper helper = new AkkaServerlessTestKitHelper<$stateClassName>();
           |
+          |  public static $testkitClassName of(Function<EventSourcedEntityContext, $entityClassName> entityFactory) {
+          |    return of("testkit-entity-id", entityFactory);
+          |  }
+          |
+          |  public static $testkitClassName of(String entityId, Function<EventSourcedEntityContext, $entityClassName> entityFactory) {
+          |    return new $testkitClassName(entityFactory.apply(new TestKitEventSourcedEntityContext(entityId)));
+          |  }
+          |
           |  public ${testkitClassName}(${entityClassName} entity) {
```

Comment:
Shall we make the constructor `private` to guide towards using the `of` factory methods (and not provide alternatives)?

Reply:
I did not because I was thinking it is worth the flexibility where users have more constructor parameters etc, but I guess they could still do that with only the `.of` factories.

Reply:
I'm thinking that the Function with context is anyway something they need to understand in the Main. If we want convenience without the Function I think it should be another `of` factory method rather than mixing constructor and factory methods.